drivers/at_cmd/at_cmd.c

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <logging/log.h>
+#include <zephyr.h>
+#include <stdio.h>
+#include <net/socket.h>
+#include <init.h>
+#include <bsd_limits.h>
+
+#include <at_cmd.h>
+
+LOG_MODULE_REGISTER(at_cmd, CONFIG_AT_CMD_LOG_LEVEL);
+
+#define RESPONSE_MAX_LEN  CONFIG_AT_CMD_RESPONSE_MAX_LEN
+#define THREAD_PRIORITY   K_PRIO_PREEMPT(CONFIG_AT_CMD_THREAD_PRIO)
+
+#define AT_CMD_OK_STR    "OK"
+#define AT_CMD_ERROR_STR "ERROR"
+#define AT_CMD_CMS_STR   "+CMS:"
+#define AT_CMD_CME_STR   "+CME:"
+
+static K_THREAD_STACK_DEFINE(socket_thread_stack, 2048);
+static K_MUTEX_DEFINE(cmd_pending);
+
+K_MSGQ_DEFINE(return_code_msq, sizeof(enum at_cmd_rc), 2, 4);
+
+static int              common_socket_fd;
+static int              last_error;
+static char             *response_buf;
+static u32_t            response_buf_len;
```

Comment:
This variable should be initialized, same for `response_buf`.
If we call `at_cmd_write`, the read Thread may fail.

Reply:
They are initialized to 0, it's given by the nature of static variables.